src/component/aurelia-form.js

```diff
@@ -1,28 +1,107 @@
-import {bindable, customElement, children} from 'aurelia-framework';
+import {bindable, customElement, children, inject} from 'aurelia-framework';
+import {Configuration} from 'aurelia-config';
 import {resolvedView} from 'aurelia-view-manager';
+import {DOM} from 'aurelia-pal';
+import {logger} from '../aurelia-form';
 
 @resolvedView('spoonx/form', 'aurelia-form')
 @customElement('aurelia-form')
+@inject(Configuration.of('aurelia-form'), DOM.Element)
 export class AureliaForm {
   @bindable behavior = '';
 
   @bindable classes = '';
 
   @bindable options = {};
 
+  @bindable validationController;
+
+  @bindable entity;
+
   @children('form-group') formGroups = [];
 
+  mapped = {};
+
+  validateTrigger;
+
+  constructor(config, element) {
+    this.config    = config;
+    this.element   = element;
+    let validation = config.validation;
+
+    if (validation) {
+      this.validationController = validation.controller.getController(this.mapped, validation.trigger);
+      this.validateTrigger      = validation.controller.getTriggers();
+    }
+  }
+
+  submit() {
+    if (!this.validationController) {
+      return;
+    }
+
+    if (!this.entity) {
+      return logger.warn('Validation on forms requires a entity to validate.');
+    }
+
+    this.validate().then(result => {
+      if (!result || !result.length) {
+        return this.emit('valid');
+      }
+
+      this.emit('invalid', result);
+    });
+  }
+
+  changed(trigger, event) {
+    let controller = this.validationController;
+
+    if (!controller) {
+      return;
+    }
+
+    let setTrigger = controller.validateTrigger;
+    let triggers   = this.validateTrigger;
+
+    if (setTrigger === triggers.manual) {
+      return;
+    }
+
+    // Specific configured, something else triggered. Return.
+    if (setTrigger < triggers.changeOrBlur && triggers[trigger] !== setTrigger) {
+      return;
+    }
+
+    this.validate(event.target.name);
+  }
+
+  validate(property) {
+    return this.validationController.validate({object: this.entity, propertyName: property});
```

Comment:
What happen if no `property` argument is provided? It will just validate the whole form?